src/azure-cli-testsdk/azure/cli/testsdk/preparers.py

```diff
@@ -140,6 +141,10 @@ def create_resource(self, name, **kwargs):
                 template += ' --allow-blob-public-access false'
             else:
                 template += ' --allow-blob-public-access true'
+            if self.allow_shared_key_access is not None and not self.allow_shared_key_access:
+                template += ' --allow-shared-key-access false'
+            else:
+                template += ' --allow-shared-key-access true'
```

Comment:
In this case `allow_shared_key_access=None` will also add `--allow-shared-key-access true`. Is it expected? How about
```suggestion
            if self.allow_shared_key_access is not None:
                template += ' --allow-shared-key-access true' if self.allow_shared_key_access else ' --allow-shared-key-access false'
```

Reply:
Agreed